File: Day15/Leetcode20.java
import java.util.Stack;

public class Leetcode20 {
    public static void main(String[] args) {
        String s="][";

        boolean result=isValidString(s);
        if(result)
        {
            System.out.println("It is valid string");
        }
        else
            System.out.println("It is not a valid string");
    }
    public static boolean isValidString(String s)
    {
        Stack <Character> st=new Stack <Character>();

        for (int i=0;i<s.length();i++)
        {
            char c=s.charAt(i);
            if(c=='['||c=='{'||c=='(')
                st.push(c);
            else
            {
                if(st.empty())
                    return false;
                char ch=st.peek();
                if((ch=='['&&c==']')||(ch=='{'&&c=='}')||(ch=='('&&c==')'))
                {
                    st.pop();
                }
                else
                    return false;
            }
        }
        return st.empty();

    }
}

File: Day15/Leetcode20.py
def isValidString(s:str)->bool:
    st=[]
    for i in s:
        if i in "([{":
            st.append(i)
        else:
            if(not st):
                return False
            else:
                ch=st[-1]
                if (ch=='(' and  i==')') or \
                    (ch=='{' and i=='}') or \
                    (ch=='[' and i==']'):
                    st.pop()
                else:
                    return False
    return not st




s="[{()]}"
result=isValidString(s)
if result:
    print("Is a valid string")
else:
    print("is not a valid string")